gocov/test.go

```diff
@@ -21,12 +21,113 @@
 package main
 
 import (
+	"bytes"
+	"fmt"
 	"io/ioutil"
 	"os"
 	"os/exec"
+	"path/filepath"
+	"sort"
+	"strings"
+	"sync"
 )
 
+func capture(wd string, args []string) ([]byte, error) {
+	cmd := exec.Command(args[0], args[1:]...)
+	cmd.Stdin = nil
+	out := &bytes.Buffer{}
+	cmd.Stdout = out
+	cmd.Stderr = out
+	cmd.Dir = wd
+	err := cmd.Run()
+
+	out2 := &bytes.Buffer{}
+	prefix := "warning: no packages being tested depend on "
+	for _, line := range strings.SplitAfter(out.String(), "\n") {
+		if !strings.HasPrefix(line, prefix) {
+			out2.WriteString(line)
+		}
+	}
+	return out2.Bytes(), err
+}
+
+func readDirNames(dirname string) []string {
+	f, err := os.Open(dirname)
+	if err != nil {
+		return nil
+	}
+	names, err := f.Readdirnames(-1)
+	_ = f.Close()
+	return names
+}
+
+// relToGOPATH returns the path relative to $GOPATH/src.
+func relToGOPATH(p string) (string, error) {
```

Comment:
I think you could do away with this function, and just run `go list -e` to get the package path for the case where the user specifies a relative package path.